pkg/controller/storagecluster/controller_test.go

```diff
@@ -943,6 +943,67 @@ func getk8sCLientWithNodesZones(
 	return k8sClient
 }
 
+func TestStorageUpgradeClusterDefaultsMaxStorageNodesPerZone(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	cluster := &corev1.StorageCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-cluster",
+			Namespace: "kube-test",
+		},
+		Status: corev1.StorageClusterStatus{
+			Phase: string(corev1.NodeInitStatus),
+		},
+	}
+	driver := testutil.MockDriver(mockCtrl)
+	/* 1 zone */
+	k8sClient := getK8sClientWithNodesZones(6, 1, cluster)
+
+	controller := Controller{
+		client: k8sClient,
+		Driver: driver,
+	}
+
+	origVersion := "2.10.0"
+	cluster.Spec.CloudStorage = &corev1.CloudStorageSpec{}
+	cluster.Spec.Image = "oci-monitor:" + origVersion
+	cluster.Spec.Version = origVersion
+	cluster.Status.Version = origVersion
+
+	expected := []*storageapi.StorageNode{}
+	for i := 0; i < 6; i++ {
+		node := storageapi.StorageNode{
+			SchedulerNodeName: "k8s-node-" + strconv.Itoa(int(i)),
+			Pools: []*storageapi.StoragePool{
+				{},
+				{},
+			},
+		}
+		expected = append(expected, &node)
```

Comment:
Can we squeeze in few nodes without storage to test that we are actually looking at the pools? Basically have a mix of storage and storageless nodes.